Octokit/Http/InMemoryCredentialStore.cs

```diff
@@ -15,7 +15,7 @@ public InMemoryCredentialStore(Credentials credentials)
 
         public Task<Credentials> GetCredentials()
         {
-            return Task.Factory.StartNew(() => _credentials);
+            return Task.FromResult(_credentials);
```

Comment:
Also, I can see a `Task.Factory.StartNew` inside `HttpClientHandler.SendAsync` that also causes contention (I see four requests enter, but none complete). 

![](http://img.pandawhale.com/67714-somethings-fucky-gif-Gah2.gif)